tbx/static_src/javascript/components/youtube-embed.js

```diff
@@ -0,0 +1,79 @@
+import Cookies from 'js-cookie';
+
+/**
+ * YouTubeConsentManager handles the consent for loading YouTube videos on a webpage.
+ */
+class YouTubeConsentManager {
+    static selector() {
+        return '.grid__embed.streamfield__embed.youtube-video-container';
+    }
+
+    /**
+     * Create a new YouTubeConsentManager.
+     */
+    constructor(node) {
+        this.consentButtonClass = 'consent-button';
+        this.dontAskAgainButtonClass = 'dont-ask-again-button';
+        this.videoPlaceholderClass = 'youtube-video-placeholder';
+        this.youtubeEmbedClass = 'youtube-embed';
+
+        this.youtubeEmbedContainer = node;
+        this.consentButton = this.youtubeEmbedContainer.querySelector(
+            `.${this.consentButtonClass}`,
+        );
+        this.dontAskAgainButton = this.youtubeEmbedContainer.querySelector(
+            `.${this.dontAskAgainButtonClass}`,
+        );
+        this.videoPlaceholder = this.youtubeEmbedContainer.querySelector(
+            `.${this.videoPlaceholderClass}`,
+        );
+        this.youtubeEmbed = this.youtubeEmbedContainer.querySelector(
+            `.${this.youtubeEmbedClass}`,
+        );
+
+        // Bind event handlers
+        this.consentButton.addEventListener(
+            'click',
+            this.handleconsentClick.bind(this),
```

Comment:
Do we need to use `bind` here? Generally we just call class methods directly with e.g. `this.handleConsentClick()`. If you change your addEventListener to use an es6 method I don't think you'll need the bind.

```this.consentButton.addEventListener('click', (e) => {...})```

Reply:
sorting in front-end ticket.